src/components/Tool/FixCommitmentInfo/FixCommitmentInfo.tsx

```diff
@@ -91,96 +128,118 @@ const FixCommitmentInfo: React.FC<Props> = ({
         doNotBatch: true,
       },
     });
+
   const [updateInvalidStatus, { loading: updating }] =
-    useUpdateInvalidStatusMutation();
-
-  const contactStatuses = contactFilterGroups?.accountList?.contactFilterGroups
-    ? (
-        contactFilterGroups.accountList.contactFilterGroups
-          .find((group) => group?.filters[0]?.filterKey === 'status')
-          ?.filters.find(
-            (filter: { filterKey: string }) => filter.filterKey === 'status',
-          ) as MultiselectFilter
-      ).options?.filter(
-        (status) =>
-          status.value !== 'NULL' &&
-          status.value !== 'HIDDEN' &&
-          status.value !== 'ACTIVE',
-      )
-    : [{ name: '', value: '' }];
-  //TODO: Make currency field a select element
-
-  const updateContact = async (
-    id: string,
-    change: boolean,
-    status?: string,
-    pledgeCurrency?: string,
-    pledgeAmount?: number,
-    pledgeFrequency?: string,
-  ): Promise<void> => {
-    const attributes = change
-      ? {
-          id,
-          status: status as StatusEnum,
-          pledgeAmount,
-          pledgeCurrency,
-          pledgeFrequency: pledgeFrequency as PledgeFrequencyEnum,
+    useUpdateStatusMutation();
+
+  const contactStatuses = useMemo(() => {
```

Comment:
Sorry I'm just seeing this now. I think you can use `getLocalizedContactStatus` and the `StatusEnum` to get a list of all the statuses rather than making an extra query to `contactFilterGroups`. `StatusEnum` should already have NULL, HIDDEN, and ACTIVE removed.